### PR TITLE
feat: WS-31695: Paychex | Add pagination support on the paychex gem so that companies can be fetch few at a time

### PR DESCRIPTION
**Pull request by suyogbaldota-workstream**
- Updated `linked_companies` to query companies with pagination parameters and return the total company records

### CODE REVIEW

File: lib/paychex/client/companies.rb [hunk 1]
```diff
@@ -85,5 +115,13 @@ def details_by_display_ids(display_ids)
         ret
       end
     end
+
+    def companies(options)
```
Comment (Prajna) [inline lib/paychex/client/companies.rb:119]: In this case, I think we could mark `companies` as private?

File: spec/paychex/client/companies_spec.rb [hunk 2]
```diff
@@ -1,16 +1,20 @@
 RSpec.describe 'Paychex' do
   describe 'companies' do
     it 'linked_companies should return list' do
-      stub_get('companies').to_return(
-        body: fixture('companies/companies.json'),
-        headers: { content_type: 'application/json; charset=utf-8' }
-      )
+      stub_get('companies?limit=100&offset=0').
```
Comment (Prajna) [inline spec/paychex/client/companies_spec.rb:4]: Would you please add another test case to cover `multiple pages`?

File: lib/paychex/client/companies.rb (hunk 1 above)
Comment (suyogbaldota-workstream) [reply lib/paychex/client/companies.rb:121]: Done. Thank you.

File: spec/paychex/client/companies_spec.rb (hunk 2 above)
Comment (suyogbaldota-workstream) [reply spec/paychex/client/companies_spec.rb]: Updated the test case since the `linked_companies` will now return paginated response only.

**Review by Prajna — APPROVED** [review]

**Review by YalinZhang1 — APPROVED** [review]

**Review by mayankdedhia — APPROVED** [review]

**Pull request merged**